keystore/src/connection/platform/wasm/storage.rs

```diff
@@ -212,20 +212,14 @@ impl WasmEncryptedStorage {
                 let transaction = idb.transaction(&[collection], TransactionMode::ReadOnly)?;
                 let store = transaction.object_store(collection)?;
                 let params = params.unwrap_or_default();
+
                 let data = store
                     .get_all(query.map(Into::into), params.limit)?
                     .await?
                     .into_iter()
-                    .filter_map(|v| {
-                        if v.is_null() || v.is_undefined() {
-                            None
-                        } else if let Ok(mut entity) = serde_wasm_bindgen::from_value::<R>(v) {
-                            entity.decrypt(&self.cipher).ok()?;
-                            Some(entity)
-                        } else {
-                            None
-                        }
-                    });
+                    .filter(|v| !v.is_null() && !v.is_undefined())
+                    .filter_map(|v| serde_wasm_bindgen::from_value::<R>(v).ok())
```

Comment:
I'm not sure we even need that initial `.filter`. If `v` is null or undefined, then serde will fail to deserialize it as an R.